folium/plugins/encoded.py

```diff
@@ -0,0 +1,120 @@
+from abc import ABC, abstractmethod
+
+from jinja2 import Template
+
+from folium.elements import JSCSSMixin
+from folium.features import MacroElement
+from folium.utilities import parse_options
+
+
+class _BaseFromEncoded(JSCSSMixin, MacroElement, ABC):
+    """Base Interface to create folium objects from encoded strings.
+
+    Derived classes must define `_encoding_type` property which returns the string
+    representation of the folium object to create from the encoded string.
+
+    Parameters
+    ----------
+    encoded: str
+        The raw encoded string from the Polyline Encoding Algorithm. See:
+        https://developers.google.com/maps/documentation/utilities/polylinealgorithm
+    **kwargs:
+        Object options as accepted by leaflet.
+    """
+
+    _template = Template(
+        """
+        {% macro script(this, kwargs) %}
+
+            var {{ this.get_name() }} = L.{{ this._encoding_type }}.fromEncoded(
+                {{ this.encoded|tojson }},
+                {{ this.options|tojson }}
+            ).addTo({{ this._parent.get_name() }});
+
+        {% endmacro %}
+        """
+    )
+
+    default_js = [
+        (
+            "polyline-encoded",
+            "https://cdn.jsdelivr.net/npm/polyline-encoded@0.0.9/Polyline.encoded.js",
+        )
+    ]
+
+    def __init__(self, encoded: str, **kwargs):
+        super().__init__()
+        self.encoded = encoded
+        self.options = parse_options(**kwargs)
```

Comment:
Instead of `parse_options`, we have a function specifically for vector classes that parses options: `path_options`. 

For PolyLine it's called like this: `path_options(line=True, **kwargs)`

For Polygon: `path_options(line=True, radius=None, **kwargs)`

Maybe good to reuse that in the child classes? It contains some sensible defaults and does some checks. Or did you not include it on purpose?